src/test/java/org/openlmis/stockmanagement/controller/StockCardControllerTest.java

```diff
@@ -172,14 +186,79 @@ public void shouldSucceedWithValidAdjustment() {
     ResponseEntity response = controller.processStock(fId, Collections.singletonList(event), request);
 
     // verify
     StockCardEntry entry = new StockCardEntry(dummyCard, StockCardEntryType.ADJUSTMENT, event.getQuantity() * -1);
     entry.setAdjustmentReason(reason);
-    verify(service).addStockCardEntries(Collections.singletonList(entry));
+    verify(stockCardService).addStockCardEntries(Collections.singletonList(entry));
     assertThat(response.getStatusCode(), is(HttpStatus.OK));
   }
+
+
+  @Test
+  public void shouldOnlyReturnEmptyLotsWhenRequested()
+  {
+    //Arbitrary values
+    Long facilityId = 1L;
+    Long productId = 2L;
+    Long stockCardId = 3L;
+    Integer numEntries = 100;
+    Boolean countOnly = false;
+
+    when(stockCardRepository.getStockCardByFacilityAndProduct(any(Long.class), any(Long.class))).thenReturn(dummyCard);
+    when(stockCardService.getStockCardById(any(Long.class), any(Long.class))).thenReturn(dummyCard);
+    when(stockCardService.getStockCards(any(Long.class))).thenReturn(new LinkedList<StockCard>(Arrays.asList(dummyCard)));
+
+
+    /* Indicate that empty Lots shouldn’t be included in stockCard.getLotsOnHand().
+       Then, below, verify that various method calls return just a single LotOnHand
+       (having stripped out the second, empty lot, that otherwise would have been included). */
+    boolean includeEmptyLots = false;
+
+
+    addLotsToStockCard(dummyCard);
+    ResponseEntity response = controller.getStockCard(facilityId, productId, numEntries, includeEmptyLots);
+    StockCard stockCard = (StockCard)response.getBody();
+    assertEquals( 1, stockCard.getLotsOnHand().size());
+
+    addLotsToStockCard(dummyCard);
+    response = controller.getStockCardById(facilityId, stockCardId, numEntries, includeEmptyLots);
+    stockCard = (StockCard)response.getBody();
+    assertEquals( 1, stockCard.getLotsOnHand().size());
+
+
+    addLotsToStockCard(dummyCard);
+    response = controller.getStockCards(facilityId, numEntries, countOnly, includeEmptyLots);
+    OpenLmisResponse openLmisResponse = (OpenLmisResponse)response.getBody();
+    List<StockCard> stockCards = (List<StockCard>)openLmisResponse.getData().get("stockCards");
+    stockCard = stockCards.get(0);
+    assertEquals( 1, stockCard.getLotsOnHand().size());
+
+
+    /* Indicate that empty Lots should be included in stockCard.getLotsOnHand().
+       Then, below, verify that various method calls return both lots (including the empty one)
+        associated with our stockCard. */
+    includeEmptyLots = true;
+
+
+    addLotsToStockCard(dummyCard);
+    response = controller.getStockCard(facilityId, productId, numEntries, includeEmptyLots);
+    stockCard = (StockCard)response.getBody();
+    assertEquals( 2, stockCard.getLotsOnHand().size());
+
+    addLotsToStockCard(dummyCard);
+    controller.getStockCardById(facilityId, stockCardId, numEntries, includeEmptyLots);
```

Comment:
Not assigning the return value of this call to response? Then code on next line will just get value of previous call.